certificate/certificate.go

```diff
@@ -240,3 +240,22 @@ func (cert *Certificate) Dump() (string, error) {
 
 	return string(result), nil
 }
+
+func (cert *Certificate) NewCertForHost(pKey string, ref string, role string) (core.Certificate, error) {
+	newCert := Certificate{
+		MajorityRule:        cert.MajorityRule,
+		MinRoles:            cert.MinRoles,
+		PublicKey:           pKey,
+		Reference:           ref,
+		PulsarPublicKeys:    cert.PulsarPublicKeys,
+		Role:                role,
+		BootstrapNodes:      make([]BootstrapNode, len(cert.BootstrapNodes)),
+		RootDomainReference: cert.RootDomainReference,
+	}
+	for i := range cert.BootstrapNodes {
```

Comment:
You can use `for i, node := range cert.BootstrapNodes` and use `node` instead of `cert.BootstrapNodes[i]`